File: endpoints/src/main/java/com/cromoteca/samples/endpoints/ExternalEntitiesEndpoint.java
package com.cromoteca.samples.endpoints;

import com.cromoteca.samples.Simple;
import com.vaadin.flow.server.auth.AnonymousAllowed;
import dev.hilla.Endpoint;
import java.util.List;
import org.springframework.data.domain.Page;
import reactor.core.publisher.Flux;

@Endpoint
@AnonymousAllowed
public class ExternalEntitiesEndpoint {
  public void page(Page<List<String>> argument) {}

  public void flux(Flux<List<String>> argument) {}

  public void inner(Inner<String> argument) {}

  public void simple(Simple argument) {}

  public static class Inner<T> {
    public Page<T> page;
    public Flux<T> flux;
  }
}

File: generator/src/test/resources/com/cromoteca/samples/endpoints/ExternalEntitiesEndpoint.ts
import client from '../../../../connect-client.default.js';
import type Page from '../../../../org/springframework/data/domain/Page.js';
import type Flux from '../../../../reactor/core/publisher/Flux.js';
import type Simple from '../Simple.js';
import type Inner from './ExternalEntitiesEndpoint/Inner.js';
import { type EndpointRequestInit } from '@vaadin/hilla-frontend';

async function flux(argument: Flux, init?: EndpointRequestInit): Promise<void> {
    return client.call('com.cromoteca.samples.endpoints.ExternalEntitiesEndpoint', 'flux', { argument }, init);
}

async function inner(argument: Inner<string>, init?: EndpointRequestInit): Promise<void> {
    return client.call('com.cromoteca.samples.endpoints.ExternalEntitiesEndpoint', 'inner', { argument }, init);
}

async function page(argument: Page<Array<string>>, init?: EndpointRequestInit): Promise<void> {
    return client.call('com.cromoteca.samples.endpoints.ExternalEntitiesEndpoint', 'page', { argument }, init);
}

async function simple(argument: Simple, init?: EndpointRequestInit): Promise<void> {
    return client.call('com.cromoteca.samples.endpoints.ExternalEntitiesEndpoint', 'simple', { argument }, init);
}

const ExternalEntitiesEndpoint = {
    flux,
    inner,
    page,
    simple,
};

export default ExternalEntitiesEndpoint;
